backend/services/activity_to_user_service.py

```diff
@@ -2,21 +2,24 @@
 from models.user import User
 from models.activity import Activity
 from repository.activity_to_user_repository import ActivityToUserRepository
+from repository.activity_repository import ActivityRepository
 
 class ActivityToUserService(BaseService):
     _repo = ActivityToUserRepository()
+    _activity_repo = ActivityRepository()
 
-    def join_activity(self, username, activity_id):
-        aTusr = ActivityToUser(activity_id = activity_id, username = username)
-        self._repo.add(aTusr)
-        return aTusr
-
-    def get_activities_with_username(self, username):
-        entries = self._repo.get_where(username)
-        return entries
-
-
-
-
+    def get_activity_to_user(self, username, activity_id):
+        result = self._repo.get_with_many(User.username == username, Activity.activity_id == activity_id)
+        return result
 
+    def join_activity(self, username, activity_id):
+        if(get_activity_to_user(username, activity_id)):
```

Comment:
If you do not find an activity the result of this get will be None. If(None) is False, but you need it to be true so you can create the ActivityToUser.
Please add a check for None https://bobbyhadz.com/blog/python-check-if-none